web/integration_test.go

```diff
@@ -236,7 +236,59 @@ func TestIntegration_StartAt(t *testing.T) {
 	cltest.CreateJobRunViaWeb(t, app, j)
 }
 
-func TestIntegration_ExternalAdapter(t *testing.T) {
+func TestIntegration_ExternalAdapter_RunLogInitiated(t *testing.T) {
+	t.Parallel()
+
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+
+	eth := app.MockEthClient()
+	logs := make(chan types.Log, 1)
+	eth.RegisterSubscription("logs", logs)
+	newHeads := make(chan models.BlockHeader, 10)
+	eth.RegisterSubscription("newHeads", newHeads)
+	app.Start()
+
+	eaValue := "87698118359"
+	eaExtra := "other values to be used by external adapters"
+	eaResponse := fmt.Sprintf(`{"data":{"value": "%v", "extra": "%v"}}`, eaValue, eaExtra)
+	mockServer, ensureRequest := cltest.NewHTTPMockServer(t, 200, "POST", eaResponse)
+	defer ensureRequest()
+
+	bridgeJSON := fmt.Sprintf(`{"name":"randomNumber","url":"%v","defaultConfirmations":10}`, mockServer.URL)
+	cltest.CreateBridgeTypeViaWeb(t, app, bridgeJSON)
+	j := cltest.FixtureCreateJobViaWeb(t, app, "../internal/fixtures/web/log_initiated_bridge_type_job.json")
+
+	logBlockNumber := 1
+	logs <- cltest.NewRunLog(j.ID, cltest.NewAddress(), logBlockNumber, `{}`)
+	time.Sleep(100 * time.Millisecond)
+	jrs, err := app.Store.JobRunsFor(j.ID)
+	assert.Nil(t, err)
+	jr := jrs[0]
+	assert.True(t, jr.Status.PendingConfirmations())
+
+	newHeads <- models.BlockHeader{Number: cltest.BigHexInt(logBlockNumber + 1)}
+	time.Sleep(100 * time.Millisecond)
+	assert.True(t, jr.Status.PendingConfirmations())
+	newHeads <- models.BlockHeader{Number: cltest.BigHexInt(logBlockNumber + 9)}
+	time.Sleep(100 * time.Millisecond)
+	assert.True(t, jr.Status.PendingConfirmations())
+	newHeads <- models.BlockHeader{Number: cltest.BigHexInt(logBlockNumber + 10)}
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
I, as I'm sure you all are as well, am pretty conflicted about the numerous sleeps in this integration test. I do understand that it's deep in the `app` `JobRunner` and that the status fluctuates from `PendingConfirmations` to `InProgress` and back to `PendingConfirmations`. 

But perhaps there's a way for us to attach a listener to our `app`'s job event runner so that we can be notified about a job completion, and not set this precedent of `time.Sleep` moving forward. Let me think on it.

Reply:
Good catch, we intended to clean those up and got carried away with writing a couple other tests and forgot to circle back. Those should be able to be handled by `Consistently`/`Eventually` checks.